pymc/gp/hsgp_approx.py

```diff
@@ -315,12 +383,17 @@ def prior_linearized(self, Xs: TensorLike):
             # First mutate the data X,
             x_new = np.linspace(-10, 10, 100)
             with model:
-                model.set_data("X", x_new[:, None])
+                pm.set_data({"X": x_new[:, None]})
 
             # and then make predictions for the GP using posterior predictive sampling.
             with model:
                 ppc = pm.sample_posterior_predictive(idata, var_names=["f"])
         """
+        # Important: fix the computation of the mean. If X is mutated later,
+        # the training mean will be subtracted, not the testing mean.
+        if self._X_mean is None:
+            self._X_mean = pt.mean(Xs, axis=0).eval()
```

Comment:
Hate to delay, and OK if it waits till a later PR, but it'd be better to use the midpoint of X here instead of the mean.  So, 
```python
self._X_center = (pt.max(Xs, axis=0) - pt.min(Xs, axis=0)) / 2
```

Reply:
On the contrary, it's great to have your input @bwengals !
Why is that though? For `S` I see why, but for the eigen vectors, all the examples before were using the mean, not the center if I'm not mistaken 🤔

Reply:
yah they were.  I think for the same reason as S, so those line up.  Then the accuracy of the approximation doesn't depend on the density of the x locations.

Reply:
Wow ... this is an interesting point! Imagine X consists of many points around zero and a single point at 100. If we want the box dimension `L` to be as small as possible (that is the whole aim right?) then we should not take the mean but the mid-point indeed 🤔

Reply:
> Even better that we're doing the transformation under the hood now, otherwise we'd have to change a lot of examples... 

Definitely, awesome that you got that working now